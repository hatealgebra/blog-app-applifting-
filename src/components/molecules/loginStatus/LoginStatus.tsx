import React from 'react';
import { RiLogoutBoxFill } from '@react-icons/all-files/ri/RiLogoutBoxFill';
import { RiAdminFill } from '@react-icons/all-files/ri/RiAdminFill';
import { AdminLinks, NavLinks } from '../../../utils/contants';
import Avatar from '../../atoms/avatar/Avatar';
import StyledLink from '../../atoms/links/link.styled';
import {
  AvatarButtonStyled,
  LoginStatuMenuStyled,
  LoginStatusContainerStyled,
} from './loginStatus.styled';

import { useAppDispatch } from '../../../store/hooks';
import { logoutAction } from '../../../store/slices/auth.slices';
import useClickOutside from '../../../hooks/useClickOutside';

const LoginStatus = () => {
  const [isOpen, setIsOpen] = React.useState(false);
  const dispatch = useAppDispatch();

  const ref = React.useRef(null);
  useClickOutside(ref, setIsOpen);

  const logOut = () => {
    dispatch(logoutAction);
  };

  return (
    <LoginStatusContainerStyled ref={ref}>
      <AvatarButtonStyled
        onClick={() => setIsOpen((prevState) => !prevState)}
        aria-label="avatar-button"
      >
        <Avatar size="md" />
      </AvatarButtonStyled>
      <LoginStatuMenuStyled
        aria-label="admin-menu"
        isOpen={isOpen}
        onClick={() => setIsOpen(false)}
      >
        <StyledLink to={AdminLinks.MY_ARTICLES}>
          <RiAdminFill />
          <span>My Articles</span>
        </StyledLink>
        <StyledLink onClick={logOut} to={NavLinks.INDEX}>
          <RiLogoutBoxFill />
          <span>Log Out</span>
        </StyledLink>
      </LoginStatuMenuStyled>
    </LoginStatusContainerStyled>
  );
};

export default LoginStatus;
