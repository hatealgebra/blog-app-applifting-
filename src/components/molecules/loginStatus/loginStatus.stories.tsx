import React from "react";
import LoginStatus from "./LoginStatus";

export const LoginStatusExample = <LoginStatus />;

export default {
  title: "Molecules/Login Status",
  component: LoginStatus,
};
