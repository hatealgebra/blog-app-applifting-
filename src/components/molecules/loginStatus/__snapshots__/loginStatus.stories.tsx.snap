// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Login Status LoginStatusExample smoke-test 1`] = `
<div class="sc-ivDvhZ conbia">
  <button aria-label="avatar-button"
          class="sc-fLBbxL bMMNBz"
  >
    <img class="sc-kDnyCx enRCkM avatar"
         src="static/media/src/images/avatar-fallback.png"
         alt="fallback user avatar"
    >
  </button>
  <ul aria-label="admin-menu"
      class="sc-bBbNsw kKnMN"
  >
    <a class="sc-eLSjS cwmAMq"
       href="/admin/my-articles"
    >
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 24 24"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path fill="none"
                d="M0 0h24v24H0z"
          >
          </path>
          <path d="M12 14v8H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm9 4h1v5h-8v-5h1v-1a3 3 0 0 1 6 0v1zm-2 0v-1a1 1 0 0 0-2 0v1h2z">
          </path>
        </g>
      </svg>
      <span>
        My Articles
      </span>
    </a>
    <a class="sc-eLSjS cwmAMq"
       href="/"
    >
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 24 24"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path fill="none"
                d="M0 0h24v24H0z"
          >
          </path>
          <path d="M5 2h14a1 1 0 0 1 1 1v18a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1zm4 9V8l-5 4 5 4v-3h6v-2H9z">
          </path>
        </g>
      </svg>
      <span>
        Log Out
      </span>
    </a>
  </ul>
</div>
`;
